From bbc6b24abdd95a7d9dfe9855630e6f0f27f48e54 Mon Sep 17 00:00:00 2001
From: Olaf Bergmann <bergmann@tzi.org>
Date: Wed, 26 Feb 2020 12:49:38 +0100
Subject: [PATCH] unit tests: Make global variables static

As of gcc 10, -fno-common is default to check for multiple
definitions of global variables at link time. This change makes
the global variables in the unit test files static to avoid
error messages caused by this.

Fixes #464
---
 tests/test_error_response.c | 4 ++--
 tests/test_wellknown.c      | 8 ++++----
 2 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/tests/test_error_response.c b/tests/test_error_response.c
index 8eb113ed4..2f3bc46c4 100644
--- a/tests/test_error_response.c
+++ b/tests/test_error_response.c
@@ -16,8 +16,8 @@
 #include <stdlib.h>
 #include <string.h>
 
-coap_pdu_t *pdu;              /* Holds the request PDU for most tests */
-coap_opt_filter_t opts;              /* option filter used for generating responses */
+static coap_pdu_t *pdu;          /* Holds the request PDU for most tests */
+static coap_opt_filter_t opts;   /* option filter used for generating responses */
 
 /************************************************************************
  ** PDU decoder
diff --git a/tests/test_wellknown.c b/tests/test_wellknown.c
index 5056b44fa..1f1cc71f5 100644
--- a/tests/test_wellknown.c
+++ b/tests/test_wellknown.c
@@ -22,10 +22,10 @@
 #define TEST_PDU_SIZE 120
 #define TEST_URI_LEN    4
 
-coap_context_t *ctx;           /* Holds the coap context for most tests */
-coap_pdu_t *pdu;           /* Holds the parsed PDU for most tests */
-coap_session_t *session;   /* Holds a reference-counted session object
-                            * that is passed to coap_wellknown_response(). */
+static coap_context_t *ctx;       /* Holds the coap context for most tests */
+static coap_pdu_t *pdu;           /* Holds the parsed PDU for most tests */
+static coap_session_t *session;   /* Holds a reference-counted session object
+                                   * that is passed to coap_wellknown_response(). */
 
 static void
 t_wellknown1(void) {
